src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceBackgroundHighlighter.java

```diff
@@ -277,6 +277,18 @@ public void onAttachOrDetach(AttachEvent event)
             documentChangedHandler_ = null;
          }
       }
+      else
+      {
+         handlers_ = new HandlerRegistrations(
```

Comment:
Should we also remove any existing handlers here as well, just in case? (Since we do initialize handlers in the constructor here: https://github.com/rstudio/rstudio/blob/e2be2eea8c7bf0852bac78034dbddfa5d71be245/src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceBackgroundHighlighter.java#L168-L170)

Or, maybe we can just remove that initialization and only init in attach / detach as you currently have things?

Reply:
We can't remove the constructor's initialization because otherwise this function won't be set as the editors' attach handler and doesn't get called. I updated the code to remove handlers regardless of attaching or detaching.